dd-java-agent/agent-logging/src/main/java/datadog/trace/logging/PrintStreamWrapper.java

```diff
@@ -0,0 +1,67 @@
+package datadog.trace.logging;
+
+import datadog.trace.api.Platform;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.nio.file.Path;
+
+public class PrintStreamWrapper extends PrintStream {
+  private static final int lineSeparatorLength = Platform.isWindows() ? 2 : 1;
+  private boolean captureOutput = false;
+  private int currentSize = 0;
+  private PrintStream printStream = null;
+
+  public PrintStreamWrapper(PrintStream ps) {
+    super(ps);
+  }
+  // use for tests only
+  public OutputStream getMainPrintStream() {
+    return super.out;
+  }
+
+  @Override
+  public void println(String x) {
+    super.println(x); // log as usual
+    if (captureOutput) {
+      currentSize += x.length() + lineSeparatorLength;
+      if (currentSize < LogReporter.MAX_LOGFILE_SIZE_BYTES) {
+        synchronized (this) {
+          printStream.println(x);
+        }
+      } else {
+        captureOutput = false;
+      }
```

Comment:
Theoretically this could lead to `currentSize` overflowing and wrapping round.

It's also not safe to use `+=` on multiple threads without synchronization.
```suggestion
      int outputLength = x.length() + lineSeparatorLength;
      if (currentSize + outputLength < LogReporter.MAX_LOGFILE_SIZE_BYTES) {
        synchronized (this) {
          printStream.println(x);
          currentSize += outputLength;
        }
      } else {
        captureOutput = false;
      }
```